keras_nlp/samplers/__init__.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tensorflow import keras
+
+from keras_nlp.samplers.greedy import Greedy
+
+
+def serialize(sampler):
```

Comment:
How is this different than using the `keras.utils.register_keras_serializable` decorator? Why are we reinventing the wheel?

Reply:
I think we want both here actually.

These utils match the helpers that exist for initializers.
https://github.com/keras-team/keras/blob/v2.11.0/keras/initializers/__init__.py#L136-L154

But we should also toss a `keras.utils.register_keras_serializable` on all our classes (base Sampler and Greedy).

Reply:
I am not sure if we want `keras.utils.register_keras_serializable`, since `Sampler` is not a tf.Module (and does not need to be). Checking the intializers code, they are not registered as serializable: https://github.com/keras-team/keras/blob/v2.11.0/keras/initializers/initializers_v2.py#L216-L268

Reply:
I did some checks, in core Keras classes not subclassing `tf.Module` or `Autotrackable` does not have the `register_keras_serializable ` annotation, e.g.:
- learning rate schedule: [link](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/optimizers/schedules/learning_rate_schedule.py#L99)
- activations: [link](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/initializers/initializers_v2.py#L885)

I think it's the same for samplers - having `serialize`/`deserialize`/`get` method, but don't do annotation.

Reply:
discussed offline, we add the annotation for best practice.